src/EthereumFlow.js

```diff
@@ -0,0 +1,221 @@
+     * the entire fee.
+     *
+     * @param {number | string | Long | BigNumber | Hbar} maxGasAllowance
+     * @returns {this}
+     */
+    setMaxGasAllowance(maxGasAllowance) {
+        this._maxGasAllowance =
+            maxGasAllowance instanceof Hbar
+                ? maxGasAllowance
+                : new Hbar(maxGasAllowance);
+        return this;
+    }
+
+    /**
+     * @template {Channel} ChannelT
+     * @template MirrorChannelT
+     * @param {import("./client/Client.js").default<ChannelT, MirrorChannelT>} client
+     * @returns {Promise<TransactionResponse>}
+     */
+    async execute(client) {
+        if (this._ethereumData == null) {
+            throw new Error(
+                "cannot submit ethereum transaction with no ethereum data"
+            );
+        }
+
+        const ethereumTransaction = new EthereumTransaction();
+        const ethereumTransactionDataBytes = this._ethereumData.toBytes();
+
+        if (this._maxGasAllowance != null) {
+            ethereumTransaction.setMaxGasAllowance(this._maxGasAllowance);
+        }
+
+        if (this._callDataFileId != null) {
+            if (this._ethereumData.callData.length === 0) {
+                throw new Error(
+                    "call data file ID provided, but ethereum data already contains call data"
+                );
+            }
+
+            ethereumTransaction
+                .setEthereumData(this._ethereumData.toBytes())
+                .setCallData(this._callDataFileId);
+        } else if (ethereumTransactionDataBytes.length <= 5120) {
+            ethereumTransaction.setEthereumData(ethereumTransactionDataBytes);
+        } else {
+            const fileId = await createFile(
+                this._ethereumData.callData,
+                client
+            );
+
+            this._ethereumData.callData = new Uint8Array();
+
+            ethereumTransaction
+                .setEthereumData(this._ethereumData.toBytes())
+                .setCallData(fileId);
+        }
+
+        return ethereumTransaction.execute(client);
+    }
+}
+
+/**
+ * @template {Channel} ChannelT
+ * @template MirrorChannelT
+ * @param {Uint8Array} callData
+ * @param {import("./client/Client.js").default<ChannelT, MirrorChannelT>} client
+ * @returns {Promise<FileId>}
+ */
+async function createFile(callData, client) {
+    const fileId = /** @type {FileId} */ (
+        (
+            await (
+                await new FileCreateTransaction()
+                    .setContents(
+                        callData.subarray(0, Math.min(callData.length, 4096))
+                    )
+                    .execute(client)
+            ).getReceipt(client)
+        ).fileId
+    );
+
+    if (callData.length > 4096) {
+        await (
+            await new FileAppendTransaction()
+                .setFileId(fileId)
+                .setContents(callData.subarray(4096))
+                .execute(client)
+        ).getReceipt(client);
+    }
+
+    return fileId;
+}
```

Comment:
Following commit f8836227c219a20661a0fdb377426a381ce08321, unit tests weren't passing. Adding
```
CACHE.ethereumTransactionDataLegacyFromBytes = (bytes) => EthereumTransactionDataLegacy.fromBytes(bytes);
CACHE.ethereumTransactionDataEip1559FromBytes = (bytes) => EthereumTransactionDataEip1559.fromBytes(bytes);
```
to the bottom of the file here lets the tests pass again. Not sure if this is the right spot to assign these, but it seemed like the best fit to as far as I could tell.

Reply:
Heh, forgot to assign those at the bottom of their respective files. Will fix.